File: src/Domain/Entities/ParentExcuse.cs
using SchoolBehaviorSystem.Domain.Enums;

namespace SchoolBehaviorSystem.Domain.Entities;

public class ParentExcuse
{
    public int Id { get; set; }
    public int StudentId { get; set; }
    public string StudentNumber { get; set; } = "";
    public string StudentName { get; set; } = "";
    public string Grade { get; set; } = "";
    public string Class { get; set; } = "";
    public Stage Stage { get; set; }
    public string ExcuseText { get; set; } = "";
    public string Attachments { get; set; } = "";
    public string AbsenceDate { get; set; } = "";
    public DateTime SubmittedAt { get; set; } = DateTime.UtcNow;
    public string SubmittedTime { get; set; } = "";
    public string Status { get; set; } = "معلق";         // معلق / مقبول / مرفوض
    public string SchoolNotes { get; set; } = "";
    public string AccessCode { get; set; } = "";

    public Student Student { get; set; } = null!;
}

File: client/src/api/parentExcuse.ts
import api from './client';

export interface ParentExcuseRow {
  id: number;
  studentId: number;
  studentNumber: string;
  studentName: string;
  grade: string;
  class: string;
  stage: string;
  excuseText: string;
  attachments: string;
  absenceDate: string;
  submittedAt: string;
  submittedTime: string;
  status: string;
  schoolNotes: string;
  accessCode: string;
  source?: string;
  miladiDate?: string;
  day?: string;
  parentName?: string;
}

export const parentExcuseApi = {
  getAll: (stage?: string, status?: string) => {
    const params = new URLSearchParams();
    if (stage) params.set('stage', stage);
    if (status) params.set('status', status);
    return api.get(`/parentexcuse?${params.toString()}`);
  },

  getPendingCount: (stage?: string) => {
    const params = new URLSearchParams();
    if (stage) params.set('stage', stage);
    return api.get(`/parentexcuse/pending-count?${params.toString()}`);
  },

  updateStatus: (id: number, status: string, notes?: string, sendMessage?: boolean) =>
    api.put(`/parentexcuse/${id}/status`, { status, notes, sendMessage }),

  delete: (id: number) =>
    api.delete(`/parentexcuse/${id}`),

  sendCustomMessage: (id: number, message: string) =>
    api.post(`/parentexcuse/${id}/send-message`, { message }),

  verifyToken: (token: string) =>
    api.get(`/parentexcuse/public/verify?token=${token}`),

  submitExcuse: (data: { token: string; reason: string; hasAttachment: boolean; absenceDate?: string }) =>
    api.post('/parentexcuse/public/submit', data),

  generateCode: (studentNumber: string, stage: string) =>
    api.post('/parentexcuse/generate-code', { studentNumber, stage }),

  generateLink: (studentNumber: string, stage: string) =>
    api.post('/parentexcuse/generate-code', { studentNumber, stage }),
};
